### PR TITLE
Raindrops

### PR DESCRIPTION
**Pull request by amymhaddad**
Please review this file.

### CODE REVIEW

File: python/raindrops/raindrops.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+FACTORS_TO_MATCH = [3, 5, 7]
+factor_to_sound = {
```
Comment (paulghaddad) [inline python/raindrops/raindrops.py]: Uppercase this as well, it's a constant.

File: python/raindrops/raindrops.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        rain_sounds = ''
+
+        for num in factor_finder(number):
+                if num in FACTORS_TO_MATCH:
+                        rain_sounds += factor_to_sound[num]
+        return rain_sounds
+
+
+def raindrops(number):
+        return number_missing_factors(number) or number_is_in_factors_to_match(number)
```
Comment (paulghaddad) [inline python/raindrops/raindrops.py]: Well named functions...and one line. Good job extracting to functions that do _one_ thing.

File: python/raindrops/raindrops.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        numbers = [3, 5, 7]
+        missing_factors = [num for num in numbers if num not in factor_finder(number)]
+        if numbers == missing_factors: 
+                return str(number)
+
+
+def number_is_in_factors_to_match(number):
+        rain_sounds = ''
+
+        for num in factor_finder(number):
```
Comment (paulghaddad) [inline python/raindrops/raindrops.py]: List comprehension OR `reduce`. Try the latter.

File: python/raindrops/raindrops.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+
+def factor_finder(number):
+        return [factor for factor in range(1, number+1) if number % factor == 0]  
+
+
+def number_missing_factors(number):
+        numbers = [3, 5, 7]
+        missing_factors = [num for num in numbers if num not in factor_finder(number)]
```
Comment (paulghaddad) [inline python/raindrops/raindrops.py]: Excellent.

File: python/raindrops/raindrops.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
+FACTORS_TO_MATCH = [3, 5, 7]
+factor_to_sound = {
+    3: 'Pling',
+    5: 'Plang',
+    7: 'Plong'
+}
+
+
+def factor_finder(number):
```
Comment (paulghaddad) [inline python/raindrops/raindrops.py]: Not super clear on what this does...consider making a better name.

File: python/raindrops/raindrops.py (hunk 2 above)
Comment (paulghaddad) [reply python/raindrops/raindrops.py]: Consider prefacing internal functions with `_`: `_number_missing_factors`

File: python/raindrops/raindrops.py (hunk 4 above)
Comment (paulghaddad) [reply python/raindrops/raindrops.py]: You can even try your hand at `filter`, which is perfect for this.

File: python/raindrops/raindrops.py (hunk 3 above)
Comment (amymhaddad) [reply python/raindrops/raindrops.py]: @paulghaddad It doesn't appear that reduce is a great option here b/c I'm testing for membership -- not applying a function to each element in a sequence.

**Comment by amymhaddad** [comment]
@paulghaddad I made many updates to simplify and clarify the code: shortened functions with clear names and added docstrings.

**Pull request merged**